File: Labs/Java/src/Java1/ObjectsAndClasses/Trade.java
package Java1.ObjectsAndClasses;

public abstract class Trade {
    private String ID;
    private String symbol;
    private int quantity;
    private double price;

    public Trade(){}


    public Trade(String ID, String symbol, int quantity) {
        this.ID = ID;
        this.symbol = symbol;
        this.quantity = quantity;
    }

    public void setPrice(double price) {
        if (price > 0 ){
            this.price = price;
        }
        else {
            System.out.println("Price can't be negative");
        }
    }

    public double getTradeValue(){
        return quantity * price;
    }

    public abstract Double calcDividend(){}

    @Override
    public String toString() {
        return "Trade{" +
                "ID='" + ID + '\'' +
                ", symbol='" + symbol + '\'' +
                ", quantity=" + quantity +
                ", price=" + price +
                '}';
    }
}

File: Labs/Scala/src/ScalaOO/Trade.scala
package ScalaOO

abstract class Trade (ID: String, initialPrice: Double)
{
//  private var _price = initialPrice
//  def price = _price
//  def price_ (priceValue: Double) = {
//    if (priceValue >= 0 ) _price = priceValue
//  }
  override def toString = s" The trade ID is $ID, " +
    s"and price: $initialPrice"


def value():Double
//  value(3.5, 3)

  //abstract method
  def isExecutable(): Boolean


}

//Object
//object Trade {
//  def apply(id: String, symbol: String, quantity: Int, price: Double) = new Trade (id, symbol, quantity, price);
//}

//Case class
//case class Trade (id: String, symbol: String, quantity: Int, initialPrice: Double)
